test-result-summary-client/src/Build/ReleaseSummary.jsx

```diff
@@ -0,0 +1,131 @@
+        fetchBuild = await fetch(`/api/getAllChildBuilds?parentId=${parentId}&buildNameRegex=${buildNameRegex}`, {
+            method: 'get'
+        })
+        const childrenBuilds = await fetchBuild.json();
+
+        for (let testGroup of childrenBuilds) {
+            //Update report with test groups that have not succeeded
+            if (testGroup.buildResult !== "SUCCESS") {
+                let testGroupId = testGroup._id;
+                let testGroupName = testGroup.buildName;
+                let testGroupUrl = testGroup.buildUrl;
+                report += `\n[**${testGroupName}**](${testGroupUrl}) \n`;
+                if (testGroup.tests) {
+                    for (let test of testGroup.tests) {
+                        if (test.testResult === "FAILED") {
+                            let testName = test.testName;
+                            let testId = test._id;
+
+                            //get test history
+                            fetchBuild = await fetch(`/api/getHistoryPerTest?testId=${testId}&limit=100`, {
+                                method: 'get'
+                            });
+                            var history = await fetchBuild.json();
+
+                            let totalRuns = 0;
+                            let totalPasses = 0;
+                            for (let testRun of history) {
+                                totalRuns += 1;
+                                if (testRun.tests.testResult === "PASSED") {
+                                    totalPasses += 1;
+                                }
+                            }
+                            //For failed tests, add links to the deep history and possible issues list
+                            report += `${testName} => [deep history ${totalPasses}/${totalRuns} passed](http://localhost:3000/deepHistory?testId=${testId}) | `
+                                        + `[possible issues](http://localhost:3000/possibleIssues?buildId=${testGroupId}&buildName=${testGroupName}&testId=${testId}&testName=${testName}) \n`;
+                        }
+                    }
+                } else {
+                    report += `⚠️ Test Job Failed ⚠️\n`
+                } 
+            }
+        }
+        this.setState({
+            body: report
+        });
+    }
+
+    async copyCodeToClipboard(){
+        var markdownText = document.getElementById('markdown-text');
+        var range, selection;
+
+        if(document.body.createTextRange){ 
+         range = document.body.createTextRange();
+         range.moveToElementText(markdownText);
+         range.select(); 
+        }
+        else if(window.getSelection){
+          selection = window.getSelection();
+          range = document.createRange();
+          range.selectNodeContents(markdownText);
+          selection.removeAllRanges();
+          selection.addRange(range);   
+        }
+        
+        document.execCommand("copy");
```

Comment:
Please check if the copy is successful and provide feedback.

```
if (document.execCommand('Copy')) {
   <Alert message="Successfuly coppied to clipboard!" type="success" showIcon />
}
```

Reply:
I have attempted to incorporate this. Please let me know if my method looks correct. (I am new to JS which is why I was not sure of a couple of things, I am sorry for the inconvenience caused)